app/models/download_request.rb

```diff
@@ -7,7 +7,40 @@ class DownloadRequest
 
   field :auth_code, type: String
   field :file_ids # Mongo ids of study files to download
-  field :tdr_files, type: Hash, default: {} # Hash of TDR project shortnames to arrays of access urls
-  field :azul_files, type: Hash, default: {} # Hash of Azul file summaries (Project Manifests, analysis_files, etc)
+  field :tdr_files, type: String # String representation of a hash of TDR project shortnames to arrays of access urls
+  field :azul_files, type: String # String representation of a hash of Azul file summaries (Project Manifests, analysis_files, etc)
   field :user_id # User making the request
+
+  before_save :sanitize_hashes
+
+  ALLOWED_TRANSFORMS = %i[encode decode]
+
+  # This is called before the data is saved to the DB to ensure the file hashes are encoded
+  # in the way the DB can accept
+  def sanitize_hashes 
+    if self.azul_files.is_a?(Hash)
+    self.tdr_files =  self.class.transform_files(tdr_files, :encode)
```

Comment:
Indent, remove extra space:

```suggestion
      self.tdr_files = self.class.transform_files(tdr_files, :encode)
```